protocol/consistencychecks.go

```diff
@@ -210,45 +223,106 @@ func (cc *ConsistencyChecks) verifySTR(str *m.SignedTreeRoot) error {
 	}
 
 	// verify hash chain
+	// TODO: clearly verify which is the client's actual expectation.
+	// See #81
 	if (str.Epoch == cc.CurrentEpoch && bytes.Equal(cc.SavedSTR, str.Signature)) ||
 		(str.Epoch == cc.CurrentEpoch+1 && str.VerifyHashChain(cc.SavedSTR)) {
 		return nil
 	}
 	return ErrorBadSTR
 }
 
-// verifyReturnedPromise verifies the returned TB
-// included in the directory's response.
-func (cc *ConsistencyChecks) verifyReturnedPromise(tb *m.TemporaryBinding,
-	uname string, key []byte,
-	str *m.SignedTreeRoot,
-	ap *m.AuthenticationPath) error {
+// verifySignedPromise checks if the returned response is match
+// with the directory's policy. Then it calls verifyFulfilledPromise
+// to verify whether issued TBs was inserted in the directory
+// as promised. Finally, it verifies the returned promise
```

Comment:
Yay, I got it, we're still on the same track.
> if they want to check for equivocation they could verify a cached TB in the next epoch.

Yes, this is what I want to include in our library.